src/scanner/syncNftStateAtBlock.js

```diff
@@ -0,0 +1,50 @@
+const { Api } = require('@cennznet/api');
+const { saveListingData, saveWalletData, updateProcessedBlockInDB, updateFinalizedBlock } = require('./utils');
+const mongoose = require('mongoose');
+require("dotenv").config();
+const logger = require('../logger');
+
+async function main () {
+    // Create the API and wait until ready
+    const api = await Api.create({network: 'nikau',});
+
+    const connectionStr = `mongodb://${process.env.MONGO_INITDB_ROOT_USERNAME}:${process.env.MONGO_INITDB_ROOT_PASSWORD}@${process.env.MONGO_HOST}:${process.env.MONGO_PORT}/admin`;
+    await mongoose.connect(connectionStr);
+    // get current block
+    let signedBlock = await api.rpc.chain.getBlock();
```

Comment:
shouldn't we use smth like `api.rpc.chain.getFinalizedHead()`?

Reply:
getfinalizedHead gives me blockHash... I will use 
let signedBlock = await api.derive.chain.bestNumberFinalized();